CONTRIBUTING.rst

```diff
@@ -0,0 +1,129 @@
+======================
+Contributing to DWI_ML
+======================
+
+Welcome to DWI_ML ! We're excited you're here and want to contribute.
+
+This article documents how to contribute improvements to DWI_ML.
+
+The development process
+=======================
+
+When writing code, please pay attention to the following aspects:
+
+Git Hooks
+---------
+
+A number of git hooks have been developed to fire some actions when certain git
+actions occur. In order to set up these hooks, it is necessary to run::
+
+    utils/gitsetup/setup-hooks.sh
+
+
+Testing
+-------
+
+We use `pytest`_ to write the tests of the code.
+
+If you are adding code into a module that already has a test file, add
+additional tests into the respective file.
+
+Coverage
+--------
+
+New contributions are required to have as close to 100% code coverage as
+possible. This means that the tests written cause each and every statement in
+the code to be executed, covering corner-cases, error-handling, and logical
+branch points.
+
+When running::
+
+    coverage run -m pytest -s --doctest-modules --verbose dwi_ml
+
+You will get the usual output of `pytest`_, but also a table that indicates the
+test coverage in each module: the percentage of coverage and also the lines of
+code that are not run in the tests. You can also see the test coverage in the
+corresponding CI build of the pull request (PR).
+
+If your contributions are to a single module, you can see test and
+coverage results for only that module without running all of the DWI_ML
+tests. For example, if you are adding code to ``dwi_ml/core/utils.py``,
+you can use::
+
+    coverage run --source=dwi_ml.core.utils -m pytest -s --doctest-modules --verbose dwi_ml/core/tests/test_utils.py
+
+You can then use ``coverage report`` to view the results, or use
+``coverage html`` and open ``htmlcov/index.html`` in your browser for a nicely
+formatted interactive coverage report.
+
+Contributions to tests that extend test coverage in older modules that are not
+fully covered are very welcome !
+
+Code style
+----------
+
+Code contributions should be formatted according to the `DWI_ML Coding Style Guideline <./doc/devel/coding_style_guideline.rst>`_.
+Please, read the document to conform your code contributions to the DWI_ML
```

Comment:
I have a small last comment. Would you mind moving this (`./doc/devel/coding_style_guideline.rst`) out of the "doc" folder?   `./coding_style_guideline.rst`
Right now, when building the html, it warns that this page is not included in any toctree, and I fear I will always be wondering why. I think it could be moved at the same place as this page (contributing).

Reply:
I'd vote for living with that until we find a solution to address that in another PR while keeping it in the current folder. You may want to open an issue as a reminder. Having all doc-related files within one folder avoids the potential clutter of the root folder.